### DISCUSSION TITLE
Package roadmap, PDF export?

### DISCUSSION THREAD

**Discussion by weilies** [dolanmiu/docx#987] [Ideas / Requests]
Hi Dolan,

Thanks for deliver such a great package. It's helpful for my next letter generator project. Is there a plan for this package to OUTPUT as pdf? Currently i am using jspdf, hoping to look for a futuristic package can generate both pdf and docx in a single code base to reduce code maintenance.

**Comment by dolanmiu** [comment]
Sorry, this does not support PDF generation

I would need to do some investigation on it, but if the file structure of `pdf` and `docx` is substantially different, then probably no.

What I do is I pass the generated `docx` file from this library into https://online2pdf.com/convert-docx-to-pdf

**Comment by daverupp** [comment]
@weilies & @dolanmiu 
Is it possible you parse an existing .docx with docx.js and then use another library to generate a pdf out of it?

my goal here is, to show a word document inside the browser - I have a client side encrypted system and would like to support previews of word documents.

thx for a hint here. 
cheers dave

**Reply by dolanmiu** [reply]
That is possible yes, there are APIs out there which convert docx to pdf

**Reply by daverupp** [reply]
@dolanmiu 
is there a way to show an existing word document in the browser with your docx.js library without going through pdf conversion. ?

**Reply by dolanmiu** [reply]
Not possible im afraid, that would require building a renderer or something
